### DISCUSSION TITLE
A few questions

### DISCUSSION THREAD

**Discussion by skelawsky** [glideapps/glide-data-grid#300] [Q&A] [answered]
I am currently looking for a suitable library to use in my project. Found yours on a reddit post and it looks great at first glance.
I have looked at the storybook and the performance is very high. Good job!

I have a few questions.
1. Is it possible to add functionality to cells? More precisely, on right click opens a custom menu, e.g. with copy, export, print options.
2. How would you solve server-side data filtering with lazy loading? Are [menus](https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-docs--menus) suitable for that or it is better to create for example custom modal with filters.
3. Where to find additional information about the 'dropdown field' from extra packages. Can I implement this in conjunction with e.g. react-select? Several columns will be able to choose one of a certain number of values, e.g. country of origin or seller from a database list. [Dropdown storybook example](https://glideapps.github.io/glide-data-grid/?path=/story/extra-packages-cells--custom-cells) does not work. Is it just a storybook bug?
4. Is it possible to add a datepicker to a cell?
5. I may need pagination. Is the implementation limited to providing data in pages and will it have any disadvantages?

**Comment by jassmith** [comment, marked as answer]
1. Yes, there are callbacks in the grid for doing that
2. I would use menus and send the sort/filter to the server. We have a basic async data source which you could extend modify for your needs. I would use it as a starting point and go from there.
3. It actually is implemented with react-select. The storybook doesn't save the result because I was lazy and didn't implement a read/write data source for that demo. It just prints to console what the selected value was.
4. There is another person doing this, maybe they will share their source with you. #287
5. Provide data in pages and set the `rowMarkerStartIndex` if you want to show the true index. There is no real downside to doing this.

**Reply by skelawsky** [reply]
Thanks for the quick reply and succinct answers. 
Is there any way to preview the code from the storybook from the 'data editor demos' tab?

**Reply by jassmith** [reply]
I didn't write it to be pretty, but here ya go: https://github.com/glideapps/glide-data-grid/blob/main/packages/core/src/data-editor/stories/data-editor-beautiful.stories.tsx
